### DISCUSSION TITLE
Ways to improve and a retrospective

### DISCUSSION THREAD

**Discussion by DeG3O** [Sau1707/ModernBot#56] [Ideas]
Hello there,

I'm coming to you with the story of the last months of using this wonderfull idea.
From the start i will disclose that i'm NOT a programmer but i've read and understood  most of the code, i have a very large base in math and C/C++.

So with that out of the way, let's see. I will start with the annoying part, namely the "AutoTrain Module".
The problem here is the way that the code is handeling the use of "spells". 
From what i've understand the recruiting process is based on a list, from naval units to land units. I'm asking, isn't there a way to create some checks to see if "Call of the ocean" is active on the city before starting the recruiting of naval units ? (ofc with tag "SPELL" active). Then the problem that i consider annoying when it wants to recruit land units and "Call of the ocean" is active, here a check for "Spartan Training or Population Grown" can't be added ?? , since the code considers "Call of the ocean" as valid to start recruiting land units, sadly is a big waste... Also an option for auto-cast before executing a command would be nice.

As for this idea, i will continue with "AntiRage" module.
1) As for now the auto-re-cast of god spells on attack can be used only on ONE attack. In improvement here is if can be set up for multiple attacks, with spells from all the gods.
2) A new option like "enchanted rage" can be very usefull, namely to spam cast "Purification with City Protection" and another with "Purification with Narcissism" on a selected city.

From what i've read in the code, there is an "AutoTrade" module to trade between own citys.
TBH i'm using some other "tools" along side ModernBot, one with an trade option and the combination between them is wonderfull. Are there any plans or ideas to creat a integrate AutoTrade in the ModernBot itself ?

This is what i've experienced in the last months with ModernBot.
That being said, hope for the best for the author and contributors of this idea, Cheers.

**Comment by Sau1707** [comment]
mhmm I don't get notifications on those...strange
